src/explorer/trialApp/TrialAppTreeItem.ts

```diff
@@ -96,6 +99,9 @@ export class TrialAppTreeItem extends SiteTreeItemBase implements ISiteTreeItem
     }
 
     public async loadMoreChildrenImpl(_clearCache: boolean, _context: IActionContext): Promise<AzExtTreeItem[]> {
+        if (isNaN(this.minutesLeft)) {
```

Comment:
I think we should just have a property on the trialAppTreeItem that is a boolean called `isExpired`.  You'll have to update it in refres, but then you don't have to have that isNaN check everywhere.

Reply:
I added it to `TrialAppClient`. I think that's the best place for it. I need to wait about 30 minutes to test it...